src/redux/sagas/utils/metadataChangelog.ts

```diff
@@ -46,8 +47,10 @@ export const getUpdatedColonyMetadataChangelog = (
       transactionHash,
       newDisplayName: newDisplayName ?? metadata.displayName,
       oldDisplayName: metadata.displayName,
-      // @TODO: Actually check whether the avatar has changed
-      hasAvatarChanged: false,
+      hasAvatarChanged:
+        newAvatarImage === undefined
+          ? false
+          : newAvatarImage !== metadata.avatar,
```

Comment:
So the logic in this util method stores the old/new values in an log entry, the returns the log. And it's up to whatever other function or component to handle the comparison logic (that's how I understand it anyway, feel free to check with @jakubcolony to confirm)

So the way you did it, breaks this pattern as just storing the comparison directly rather then the old/new values.

Reply:
Well as far as I have checked, this changelog is storing the old and new values of the colony name but not of the avatar, only if the avatar changed by the time this changelog was emitted (`hasAvatarChanged`). I'll wait for @jakubcolony to come back and throw some light into this tho